drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c

```diff
@@ -0,0 +1,488 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_mipi_dbi_flexio_lcdif
+
+#include <zephyr/drivers/dma.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/mipi_dbi.h>
+#include <zephyr/drivers/misc/nxp_flexio/nxp_flexio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <fsl_edma.h>
+#include <fsl_flexio_mculcd.h>
+
+LOG_MODULE_REGISTER(display_mcux_flexio_lcdif, CONFIG_DISPLAY_LOG_LEVEL);
+
+struct stream {
+	const struct device *dma_dev;
+	uint32_t channel; /* stores the channel for dma */
+	struct dma_config dma_cfg;
+	struct dma_block_config dma_blk_cfg;
+};
+
+struct mcux_flexio_lcdif_config {
+	FLEXIO_MCULCD_Type *flexio_lcd_dev;
+	const struct device *flexio_dev;
+	const struct pinctrl_dev_config *pincfg;
+	const struct nxp_flexio_child *child;
+	uint32_t baudrate_bps;
+	uint8_t data_bus_width;
+	/* Reset GPIO */
+	const struct gpio_dt_spec reset;
+	const struct gpio_dt_spec cs_gpio;
+	const struct gpio_dt_spec rs_gpio;
+	const struct gpio_dt_spec rdwr_gpio;
+};
+
+struct mcux_flexio_lcdif_data {
+	struct stream dma_tx;
+	struct k_sem transfer_done;
+	const struct mipi_dbi_config *active_cfg;
+};
+
+static int flexio_lcdif_isr(void *user_data)
```

Comment:
Can we omit this ISR? It seems the parent flexio driver will skip calling the ISR callback if it is `NULL`

Reply:
Thanks for catching this.